tests/unit/connector/test_http_input.py

```diff
@@ -276,6 +311,37 @@ def test_get_next_with_hmac_of_raw_message(self):
         connector_next_msg, _ = connector.get_next(1)
         assert connector_next_msg == expected_event, "Output event with hmac is not as expected"
 
+    def test_endpoint_has_credentials(self, create_credentials):
+        credential_file_path = create_credentials
+        mock_env = {"LOGPREP_CREDENTIALS_FILE": str(credential_file_path)}
+        with mock.patch.dict("os.environ", mock_env):
+            new_connector = Factory.create({"test connector": self.CONFIG}, logger=self.logger)
+            new_connector.pipeline_index = 1
+            new_connector.setup()
+            endpoint_config = new_connector.http_server.endpoints_config.get("/auth-json-secret")
+            print(endpoint_config.credentials)
+            assert endpoint_config.credentials.username
+            assert endpoint_config.credentials.password
+
+    def test_endpoint_has_basic_auth(self, create_credentials):
+        credential_file_path = create_credentials
+        mock_env = {"LOGPREP_CREDENTIALS_FILE": str(credential_file_path)}
+        with mock.patch.dict("os.environ", mock_env):
+            new_connector = Factory.create({"test connector": self.CONFIG}, logger=self.logger)
+            new_connector.pipeline_index = 1
+            new_connector.setup()
+            resp = requests.post(url=f"{self.target}/auth-json-file", timeout=0.5)
+            assert resp.status_code == 401
+            basic = HTTPBasicAuth("wrong", "credentials")
+            resp = requests.post(url=f"{self.target}/auth-json-file", auth=basic, timeout=0.5)
+            assert resp.status_code == 401
+            basic = HTTPBasicAuth("user", "file_password")
+            resp = requests.post(url=f"{self.target}/auth-json-file", auth=basic, timeout=0.5)
+            assert resp.status_code == 200
+            basic = HTTPBasicAuth("user", "secret_password")
+            resp = requests.post(url=f"{self.target}/auth-json-secret", auth=basic, timeout=0.5)
+            assert resp.status_code == 200
+
```

Comment:
```suggestion
            basic = HTTPBasicAuth("user", "password")
            resp = requests.post(
                url=f"{self.target}/auth-json-secret/AB/json", auth=basic, timeout=0.5
            )
            assert resp.status_code == 200
```

* real usage of a regex endpoint from above